src/common/datomish/pair_chan.cljc

```diff
@@ -71,6 +71,13 @@
      (consume-pair (cljs.core.async/<! ~pc-chan))
      (consume-pair (clojure.core.async/<! ~pc-chan))))
 
+(defmacro <??
+  "Takes from the channel if it's non-nil."
+  [pc-chan]
+  `(let [c# ~pc-chan]
+     (when c#
+       (datomish.pair-chan/<? c#))))
```

Comment:
Because this is a macro, `pc-chan` is a _form_ — it's a piece of syntax, not a channel. And similarly, `c#` is a gensym; not yet a thing.

The naming convention applies to what the programmer passes to `<??`, not here inside this very simple expansion, and indeed it only applies to functions — we usually call variables that _are_ channels `chan` or `c` or `pair-chan`.

So I'd use this like `(<?? (<my-channel-function 5))`, or `(let [c (<my-channel-function 5)] (println (<?? c)))`.

You can play around with this in a REPL (`lein repl`):

```
user=> (defmacro <??
  #_=>   "Takes from the channel if it's non-nil."
  #_=>   [pc-chan]
  #_=>   `(let [c# ~pc-chan]
  #_=>      (when c#
  #_=>        (datomish.pair-chan/<? c#))))
#'user/<??
user=> (macroexpand '(<?? (<my-fn 5)))
(let*
 [c__14311__auto__ (<my-fn 5)]
 (clojure.core/when c__14311__auto__ (datomish.pair-chan/<? c__14311__auto__)))
```